maas-api/internal/handlers/models_test.go

```diff
@@ -164,6 +199,40 @@ func TestListingModels(t *testing.T) {
 			assert.Equal(t, tc.expectedModel.Ready, actualModel.Ready)
 		})
 	}
+
+	// Verify model with full metadata
+	t.Run("model-with-metadata has modelDetails", func(t *testing.T) {
```

Comment:
These tests are already executed in the main loop. 

We could extend `LlmTestScenario` to support per-scenario assertion hooks - functions defined on each test case that capture scenario-specific expectations as closures. We could conditionally call them in the main loop above - https://github.com/opendatahub-io/models-as-a-service/pull/318/changes#diff-017707c5a6d5f81952d731af8e6df1e0653a282bc74129aef05431b0104954fcR189-R201